packages/react-native/Libraries/AppDelegate/RCTReactNativeFactory.mm

```diff
@@ -0,0 +1,235 @@
+#define USE_OSS_CODEGEN 0
+#endif
+
+using namespace facebook::react;
+
+@interface RCTReactNativeFactory () <RCTComponentViewFactoryComponentProvider, RCTHostDelegate>
+@end
+
+@implementation RCTReactNativeFactory
+
+- (instancetype)initWithDelegate:(id<RCTReactNativeFactoryDelegate>)delegate {
+  if (self = [super init]) {
+    [self _setUpFeatureFlags];
+    self.delegate = delegate;
+
+    RCTSetNewArchEnabled([self newArchEnabled]);
+    [RCTColorSpaceUtils applyDefaultColorSpace:[self defaultColorSpace]];
+    RCTEnableTurboModule([self turboModuleEnabled]);
+    
+    self.rootViewFactory = [self createRCTRootViewFactory];
+    
+    if ([self newArchEnabled] || [self fabricEnabled]) {
+      [RCTComponentViewFactory currentComponentViewFactory].thirdPartyFabricComponentsProvider = self;
+    }
+  }
+  
+  return self;
+}
+
+- (std::shared_ptr<facebook::react::TurboModule>)getTurboModule:(const std::string &)name
+                                                      jsInvoker:(std::shared_ptr<facebook::react::CallInvoker>)jsInvoker
+{
+  return facebook::react::DefaultTurboModules::getTurboModule(name, jsInvoker);
+}
+
+- (std::shared_ptr<facebook::react::TurboModule>)getTurboModule:(const std::string &)name
+                                                     initParams:
+                                                         (const facebook::react::ObjCTurboModule::InitParams &)params
+{
+  return nullptr;
+}
+
+- (RCTBridge *)createBridgeWithDelegate:(id<RCTBridgeDelegate>)delegate launchOptions:(NSDictionary *)launchOptions
+{
+  return [_delegate createBridgeWithDelegate:delegate launchOptions:launchOptions];
+}
+
+- (UIView *)createRootViewWithBridge:(RCTBridge *)bridge
+                          moduleName:(NSString *)moduleName
+                           initProps:(NSDictionary *)initProps
+{
+  return [_delegate createRootViewWithBridge:bridge moduleName:moduleName initProps:initProps];
+}
+
+- (RCTColorSpace)defaultColorSpace
+{
+  return [_delegate defaultColorSpace];
+}
+
+- (BOOL)newArchEnabled {
+   return _delegate.newArchEnabled;
+}
+
+- (BOOL)fabricEnabled {
+  return _delegate.fabricEnabled;
+}
+
+- (BOOL)turboModuleEnabled {
+  return _delegate.turboModuleEnabled;
+}
+
+- (BOOL)bridgelessEnabled {
+  return _delegate.bridgelessEnabled;
+}
+
+#pragma mark - RCTTurboModuleManagerDelegate
+
+- (Class)getModuleClassFromName:(const char *)name
+{
+#if RN_DISABLE_OSS_PLUGIN_HEADER
+  return RCTTurboModulePluginClassProvider(name);
+#else
+  return RCTCoreModulesClassProvider(name);
+#endif
+}
+
+- (id<RCTTurboModule>)getModuleInstanceFromClass:(Class)moduleClass
+{
+  return RCTAppSetupDefaultModuleFromClass(moduleClass);
+}
+
+#pragma mark - RCTComponentViewFactoryComponentProvider
+
+- (NSDictionary<NSString *, Class<RCTComponentViewProtocol>> *)thirdPartyFabricComponents
+{
+#if USE_OSS_CODEGEN
+  return [RCTThirdPartyComponentsProvider thirdPartyFabricComponents];
+#else
+  return @{};
+#endif
+}
```

Comment:
this should use the delegate implementation if available. Otherwise we can't provide custom fabric components

Reply:
I think that with `dependencyProvider` we don't need to expose it through delegate?

```
- (NSDictionary<NSString *, Class<RCTComponentViewProtocol>> *)thirdPartyFabricComponents
{
  return self.dependencyProvider ? self.dependencyProvider.thirdPartyFabricComponents : @{};
}
```

I've added option to forward setting RCTDependencyProvider on ReactNativeFactory.

Reply:
Damn it looks like the delegate needs to provide `dependencyProvider` - this complicates things a bit but I'll try to add it so that we can overwrite `thirdPartyFabricComponents` from AppDelegate

Reply:
Solved, you can take a look at the implementation now